Documentation/DocuBlocks/Rest/Cursors/post_api_cursor_identifier_batch.md

```diff
@@ -0,0 +1,118 @@
+
+@startDocuBlock post_api_cursor_identifier_batch
+@brief Return results from an existing cursor once more
+
+@RESTHEADER{POST /_api/cursor/{cursor-identifier}/{batch-identifier}, Read batch from cursor again, batchQueryCursorPost}
+
+@RESTURLPARAMETERS
+
+@RESTURLPARAM{cursor-identifier,string,required}
+The ID of the cursor.
+
+@RESTURLPARAM{batch-identifier,string,required}
+The ID of the batch. The first batch has an ID of `1` and the value is
+incremented by 1 with every batch. You can only request the latest batch again.
+Earlier batches are not kept on the server-side.
+
+@RESTDESCRIPTION
+You can use this endpoint to retry fetching the latest batch from a cursor.
+The endpoint requires the `retriable` query option to be enabled for the cursor.
+
+If the cursor is still alive, returns an object with a batch of query results.
+Calling this endpoint does not advance the cursor.
+
+@RESTRETURNCODES
+
+@RESTRETURNCODE{200}
+The server responds with *HTTP 200* in case of success.
+
+@RESTREPLYBODY{error,boolean,required,}
+A flag to indicate that an error occurred (`false` in this case).
+
+@RESTREPLYBODY{code,integer,required,integer}
+The HTTP status code.
+
+@RESTREPLYBODY{result,array,optional,}
+An array of result documents for the current batch.
+
+@RESTREPLYBODY{hasMore,boolean,required,}
+A boolean indicator whether there are more results available for the cursor on
+the server. `false` if this is the last batch.
+
+Note that even if `hasMore` returns `true`, the next call might still return no
+documents. If, however, `hasMore` is `false`, then the cursor is exhausted.
+Once the `hasMore` attribute has a value of `false`, the client can stop.
+
+@RESTREPLYBODY{count,integer,optional,int64}
+The total number of result documents available (only
+available if the query was executed with the `count` attribute set).
+
+@RESTREPLYBODY{id,string,optional,string}
+The ID of the cursor for fetching more result batches.
+
+@RESTREPLYBODY{extra,object,optional,}
+An optional JSON object with extra information about the query result. It can
+have the attributes `stats`, `warnings`, `plan`, and `profile` with nested
+objects, like the initial cursor response. Every batch can include `warnings`.
+The other attributes are only delivered as part of the last batch in case of a
+cursor with the `stream` option enabled.
+
+@RESTRETURNCODE{400}
+If the cursor and the batch identifier are omitted, the server responds with
+*HTTP 400*.
+
+@RESTREPLYBODY{error,boolean,required,}
+A flag to indicate that an error occurred (`false` in this case).
+
+@RESTREPLYBODY{code,integer,required,integer}
+The HTTP status code.
+
+@RESTREPLYBODY{errorNum,integer,required,int64}
+A server error number (if `error` is `true`).
+
+@RESTREPLYBODY{errorMessage,string,required,string}
+A descriptive error message (if `error` is `true`).
+
```

Comment:
These are actually *optional* attributes and not *required* attributes.
They are only set in case of errors.

Reply:
Well, this specific reply body is for the error case, where they are always set. I consider this to be an oddity of our DocuBlocks. I would agree if he had a unified description of the response structure, but then all non-error fields would need to be optional...